browser/mapping.go

```diff
@@ -434,9 +434,16 @@ func mapPage(vu moduleVU, p *common.Page) mapping {
 			})
 		},
 		"close": func(opts goja.Value) error {
-			if vu.tq != nil {
+			// There could be multiple pages and they could all need the taskqueue
+			// which means that we should only close the taskqueue once we only
+			// have one or less pages in the browserContext.
+			pp := p.Context().Pages()
+
+			if len(pp) <= 1 && vu.tq != nil {
```

Comment:
This fix will never work since the page close is asynchronous, which means we would have to work with an event loop or wait here for some time until the page count decreases by one. Neither of these are good solution. Instead i've opted to work with a `taskqueue` per page, which you will find in a subsequent commit.